src/main/java/cz/cvut/kbss/termit/rest/BaseController.java

```diff
@@ -35,7 +35,7 @@
  * <p>
  * Will be used to define general security for the public API.
  */
-@PreAuthorize("hasRole('" + SecurityConstants.ROLE_USER + "')")
+@PreAuthorize("hasRole('" + SecurityConstants.ROLE_RESTRICTED_USER + "')")
```

Comment:
I would probably rather go with `hasAnyRole` of the supported ones and not inherit the roles (it feels weird that an admin inherits restricted role, but that is probably just a naming thing).

Reply:
I was considering this option as well, mainly because in the current version inferring the roles in the backend changes makes the semantics of the role on backend different from the semantics of the role in data.

But then I decided for this as I originally anticipated many more security annotations and wanted to avoid repeating the roles many times.

Anyway, being aware of these options, I will keep the implementation as it is.